src/api/system.c

```diff
@@ -851,19 +851,30 @@ static int f_get_fs_type(lua_State *L) {
 
 
 static int f_ftruncate(lua_State *L) {
-#if LUA_VERSION_NUM < 503
-  // note: it is possible to support pre 5.3 and JIT
-  //       since file handles are just FILE*  wrapped in a userdata;
-  //       but it is not standardized. YMMV.
-  #error luaL_Stream is not supported in this version of Lua.
```

Comment:
So we'll allow compiling even if it's not supported?
I guess it's fine.

Reply:
I recall this point being raised in the last PR committee meeting. I'm still unsure about this approach, but this should allow us to have more backwards compatibility with other Lua versions.